cheroot/test/test_server.py

```diff
@@ -236,3 +237,77 @@ def test_peercreds_unix_sock_with_lookup(peercreds_enabled_server):
         peercreds_text_resp = requests.get(unix_base_uri + PEERCRED_TEXTS_URI)
         peercreds_text_resp.raise_for_status()
         assert peercreds_text_resp.text == expected_textcreds
+
+
+@pytest.mark.skipif(
+    IS_WINDOWS,
+    reason=(
+        'This regression test is for a Linux bug, and the resource module '
+        'is not available on Windows'
+    ),
+)
+@pytest.mark.forked
+def test_high_number_of_file_descriptors(http_server):
+    """Test the server does not crash with a high file-descriptor value.
+
+    This test should cause a server crash, as the server will try to
+    select() a file-descriptor higher than 1024.
```

Comment:
This is inaccurate. The test should check that it doesn't crash under the buggy conditions.
```suggestion
    This test shouldn't cause a server crash when trying to access
    file-descriptor higher than 1024.

    The earlier implementation used to rely on `select()` syscall that
    doesn't support file descriptors with numbers higher than 1024.
```